scripts/quickstart

```diff
@@ -1,94 +1,150 @@
 #!/bin/bash -e
 
-CMD="$(realpath "$0")"
-DIR="$(dirname "${CMD}")"
-BASE_DIR="$(dirname "${DIR}")"
-CONFIG_DIR="${BASE_DIR}/config"
-CERTS_DIR="${CONFIG_DIR}/certs"
-
-DOMAIN=openbalena.local
-
-usage() {
-  echo "usage: $0 [-h] [-p] [-d DOMAIN] -U EMAIL -P PASSWORD"
-  echo
-  echo "  -p           patch hosts - patch the host /etc/hosts file"
-  echo "  -d DOMAIN    the domain name this deployment will run as, eg. example.com. Default is 'openbalena.local'"
-  echo "  -U EMAIL     the email address of the superuser account, used to login to your install from the Balena CLI"
-  echo "  -P PASSWORD  the password to use for the superuser account."
-  echo
-}
+BLACK=`tput setaf 0`
+RED=`tput setaf 1`
+GREEN=`tput setaf 2`
+YELLOW=`tput setaf 3`
+BLUE=`tput setaf 4`
+MAGENTA=`tput setaf 5`
+CYAN=`tput setaf 6`
+WHITE=`tput setaf 7`
+
+BOLD=`tput bold`
+RESET=`tput sgr0`
+
+quickstart() {
+
+  # for macos machines, we need proper OpenSSL...
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+    echo -e "${BLUE}INFO: macOS detected. Bringing in OpenSSL if installed..."
+    export PATH="/usr/local/opt/openssl/bin:$PATH"
+  fi
+
+  OPENSSL_VERSION=$(openssl version -v)
+  if [[ "${OPENSSL_VERSION}" =~ ^LibreSSL.*$ ]]; then
+    echo -e "${RED}ERROR: You may not have a compatible OpenSSL version (${OPENSSL_VERSION}). Please install OpenSSL version 1.0.2q or above."
+    exit 1
+  fi
+
+  REALPATH=
+  REALPATHS=(
+    'realpath'
+    'readlink'
+    'greadlink -f'
+  )
+  for cmd in "${REALPATHS[@]}"
+  do
+    if type ${cmd%% *} &>/dev/null; then
+      REALPATH=$cmd
+      break
+    fi
+  done
+
+  if [ -z $REALPATH ]; then
+    echo "${RED}ERROR: Unable to find suitable command for realpath."
```

Comment:
Please reset the console colour before exiting.

Reply:
agreed - I will chuck a `${RESET}` in there to be sure